### ISSUE TITLE
SourceMaps

### ISSUE THREAD

**Issue by gillesgoetsch** [suomato/base-camp#17]
Hello.

Awesome Boiler!
Any way to get source maps from the generated css?

Thanks,
Gilles

**Comment by suomato** [comment]
Hi,

Try to edit `webpack.config.js` file:



```
const cssLoader = {
  loader: 'css-loader',
  options: { 
    minimize: inProduction, 
    sourceMap: true, // add this 
  },
};

module.exports = {
  devtool: false, // add this

  module: {
    rules: [
      ...
      {
        test: /\.sass$/,
        use: [
          'vue-style-loader',
          // extractCss, // comment this out
          cssLoader,
          {
            loader: 'sass-loader',
            options: {
              indentedSyntax: true,
              sourceMap: true, // add this
              includePaths: [
                path.resolve(__dirname, '../resources/assets/sass'),
              ],
              data: '@import "variables";',
            },
          },
        ],
      },
    ]
  }
  plugins: [
    ...
    new webpack.SourceMapDevToolPlugin({}), // add this
  ]
}
```

You should get source map but all styles will be in `.js`-file. I couldn't get it work with `MiniCssExtractPlugin` :(

**Comment by gillesgoetsch** [comment]
Hi @suomato 

Thanks for your super fast reply.

Got it working only with :

`module.exports = {
  devtool: 'cheap-module-eval-source-map',`

and:

`new webpack.SourceMapDevToolPlugin({
        filename: "[file].map",
        exclude: ["/vendor/"]
    }),`

Generating .css.map files.

Thanks!

**Comment by gillesgoetsch** [comment]
PurgeCss destroys the maps. Used:

`const purgeCssFiles = inProduction ? ['app'] : [''];`

and:

```
new PurgecssPlugin({
    paths: () =>
    glob.sync(path.join(__dirname, '../resources/**/*'), { nodir: true }),
    only: purgeCssFiles,
}),
```

to keep it and only do the purge in prod builds.

**Comment by suomato** [comment]
Excellent! Great solution!

**Comment by gillesgoetsch** [comment]
I also switched all sass files to scss but more importantly:

Moved the whole build (webpack part) out of the /wp-content/themes/theme/ directory into /theme/ which then copies all files and views into the theme based on the config.

This allows for a much cleaner theme folder and doesn't mix source data with generated build data for safe deploys.

If you find this relevant to be included, let me know.
